File: app/src/main/java/com/example/miwok/Word.java
package com.example.miwok;

public class Word {
    private String mDefaultword;
    private String mMivokword;
    private int mResId;
    public static final int IMAGE_PROVIDED = -1;
    public Word(String def, String mivok){
        mDefaultword = def;
        mMivokword = mivok;

    }
    public Word(String def, String mivok, int res){
        mDefaultword = def;
        mMivokword = mivok;
        mResId = res;

    }

    public String getDefaultword() {
        return mDefaultword;
    }
    public String getMivokword(){
        return mMivokword;
    }
    public int getImageId(){
        return mResId;
    }

    public boolean hasImg(){
        return mResId != IMAGE_PROVIDED;
    }
}

File: app/src/main/java/com/example/miwok/model/Word.kt
package com.example.miwok.model

class Word {
    var defaultword: String
        private set
    var mivokword: String
        private set
    var imageId = 0
        private set
    var arabicword: String
        private set

    constructor(def: String, mivok: String, arabic: String) {
        defaultword = def
        mivokword = mivok
        arabicword = arabic
    }

    constructor(def: String, mivok: String, arabic: String, res: Int) {
        defaultword = def
        mivokword = mivok
        arabicword = arabic
        imageId = res
    }

    fun hasImg(): Boolean {
        return imageId != IMAGE_PROVIDED
    }

    companion object {
        const val IMAGE_PROVIDED = -1
    }
}